dashboard/controller/assignments/exportCSV.js

```diff
@@ -0,0 +1,92 @@
+// include libraries
+var superagent = require('superagent'),
+	async = require('async'),
+	common = require('../../helper/common');
+
+module.exports =  function exportCSV(req, res) {
+	var assignment = [];
+
+	const validateAssignment=(assignment)=>{
+		const assign={
+			_id:"",
+			name:"",
+			title:"",
+			activity:"",
+			creation_time:"",
+			stroke:"",
+			fill:"",
+			instruction:"",
+			dueDate:"",
+			classroom:"",
+			isAssigned: ""
+		};
+		if(assignment._id){
+			assign._id=assignment._id;
+		}
+		if(assignment.name){
+			assign.name=assignment.name;
+		}
+		const assignedWork=assignment.assignedWork?.metadata;
+		if(assignedWork?.title){
+			assign.title=assignedWork.title;
+		}
+		if(assignedWork?.activity){
+			assign.activity=assignedWork.activity;
+		}
+		if(assignedWork?.creation_time){
+			assign.creation_time=new Date(assignedWork.creation_time);
+		}
+		if(assignedWork?.buddy_color?.stroke){
+			assign.stroke=assignedWork.buddy_color.stroke;
+		}
+		if(assignedWork?.buddy_color?.fill){
+			assign.fill=assignedWork.buddy_color.fill;
+		}
+		if(assignment.instructions){
+			assign.instruction=assignment.instructions;
+		}
+		if(assignment.dueDate){
+			assign.dueDate=new Date(assignment.dueDate);
+		}
+		if(assignment.classrooms){
+			assign.classroom='';
+			for(var i=0;i<assignment.classrooms.length;i++){
+				assign.classroom += assignment.classrooms[i]+(i!=assignment.classrooms.length-1?',':'');
+			}
+		}
+		if(assignment.isAssigned){
+			assign.isAssigned=assignment.isAssigned;
+		}
+		return assign;
+
+	}
+	async.series([
+		function(callback) {
+			superagent
+				.get(common.getAPIUrl(req) + 'api/v1/assignments')
```

Comment:
I don't think the issue is about exporting Assignments. The requirement is for exporting the Deliveries.